### DISCUSSION TITLE
Show Recently added per library on home hub

### DISCUSSION THREAD

**Discussion by JonzaUK** [pannal/plex-for-kodi#156] [Ideas]
Pull through recently added per library on the home hub, like modern plex clients, PM4K already pulls though recently added on each library hub its just combined on the Home Hub

Edit: use new API for home hub to allow recently added per library 

Recently Added Series - On Series TAB - PM4K

![Series TAB PM4K](https://github.com/user-attachments/assets/77771823-4e5e-4b2f-9cd5-939cdefa7c2c)

Recently Added Anime - On Anime TAB - PM4K
![Anime Tab - PM4K](https://github.com/user-attachments/assets/1e463d73-7259-434f-a712-bb5b98cc31bb)

Current Home Screen - Showing Recently Added TV combining individual libraries into one. PM4K
![Home Hub - PM4K](https://github.com/user-attachments/assets/62e53514-7432-430f-b274-2d85e4dfe313)

Home Screen of plex app (Apple TV but Android TV etc is the same.)

https://github.com/user-attachments/assets/d6f720a6-a0a9-49d6-9d77-e9a48d16fa32

**Comment by pannal** [comment]
> its just combined on the Home Hub

No, it's using a fully different API endpoint on Home compared to the libraries.

**Reply by pannal** [reply]
We're using the old approach via `/hubs?includeMarkers=1`.

I'm planning on looking at this more in depth in 0.8.1.

**Reply by JonzaUK** [reply]
awesome thank you <3

**Reply by pannal** [reply]
So, how would this look like?

Currently we're combining the section types into Recently Added SECTIONTYPE (TV, Movies etc.) on Home, because that's what /hubs returns..

My idea would be: You can long-press/contextmenu on a Library and set it as "Show on Home", which would then add `Recently Added in LIBRARYNAME`. Once you do that, the original/"meta" hubs are removed and you have to do this for every library you want on Home.

OnDeck/In Progress would stay the same.

Does that sound like it?

**Reply by JonzaUK** [reply]
My idea was an option in the settings "Modern Home Hub" which would just mirror modern plex showing recently added per library for all library's.

but your way would give more granular control over what's on the home screen... which might be nicer for other users... going off your way I thought maybe it would be possible to pin other hubs to the home example "Rediscover" but that might get more complicated? 

I'm happy with either way as I can achieve modern client view either way, though your way sounds like it could allow further expansion like the above further in the future.

**Reply by pannal** [reply]
I just had a quick look at how we're currently querying the hubs for the libraries (not Home). Turns out, we already have the necessary data, I just need to find a good way to compile it, while ideally providing the same customizability as other clients.

Another option would be to keep "Recently Added SECTIONTYPE" minus the library you added separately to home. When a user adds all their Libraries' Recently Added to home, this hub would just go away.

Not sure yet.

**Reply by pannal** [reply]
And then there's the question of the order. That's going to be difficult. Need further investigation.

**Reply by JonzaUK** [reply]
Could you link order to order of the library order at the top? believe that's how it works in plex.

**Reply by pannal** [reply]
That would be the easiest, sure. Missed the forest for the trees here.

Not sure if that's always desired, but it would be a good first solution.

**Reply by JonzaUK** [reply]
Sorting via Library order at the top as the default is good first solution and maybe further down the line allow the user reorder if possible.

Agree with your point, allowing a context menu option that says "add recently added to home" and either once all library sections are addedit removes the sectiontype hub or as soon as one library is added to the home is removes the sectiontype hub straight away? 

Last bit is the only bit I'm unsure of which would be the best way.

Edit: Maybe just add a setting that removes / hides the combine section view. and this would allow users to add what librarys they want on the home hub and that can include or exclude the combined hub?

**Reply by MrMookdog** [reply]
> Could you link order to order of the library order at the top? believe that's how it works in plex.

Yeah thats the way it works in plex for android. I am getting gitty at the thought of this becoming a reality lol
